### ISSUE TITLE
Invalid string encoding found in Parquet file when temp_directory is used

### ISSUE THREAD

**Issue by goober** [duckdb/duckdb#7962]
### What happens?

When I am converting larger ndjson files to parquet, some columns get corrupted when the `memory_limit` and `temp_directory` options are set.

```
CREATE TEMP TABLE t1 AS (SELECT
                             		row as id,
                             		date as time,
                             		strftime(time, '%%Y-%%m-%%d') dt,
                             		location,
                             		record.title as title,
                             		record.description as description,
                             		FROM read_json('ndjson/*.ndjson', format=newline_delimited,
										columns={
											row:'varchar',
								                         location:'varchar',
								    		        record:'struct(title varchar,  "description" varchar)',
								    		        date:'timestamp'
										})
									);

COPY (select * from t1 order by location) TO 'parquet' (FORMAT PARQUET, COMPRESSION 'ZSTD', PARTITION_BY (dt));

```

```
select * from read_parquet('parquet/dt=*/*.parquet', hive_partitioning=1) LIMIT 1;

Error: Invalid Input Error: Invalid string encoding found in Parquet file: value "\xE2\xBA\x91.y\xFD\x05\x00\x97\xA8U:y\xFD\x05\x00\xD6\x92\xF7K" is not valid UTF8!
```

The issue is not present if the `memory_limit` is not set and everything fits into memory.

The ndjson files are approximately around 200MB each, and we are trying to convert around 10 of those in the same query. 

### To Reproduce

Unfortunately I have not been able to reproduce the issue with a randomly generated dataset that can be shared. I will try to reproduce the issue with sharable data if necessary.

### OS:

Linux amd64 / OSX arm64

### DuckDB Version:

0.8.1

### DuckDB Client:

golang / cli

### Full Name:

Mathias Åhsberg

### Affiliation:

Independent

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by hannes** [comment]
Can you double check the string encodings in the ndjson files?

**Comment by goober** [comment]
Further investigation have narrowed down the issue.
If we provide a wildcard to the read_json function as stated above e.g. `read_json('ndjson/*.ndjson',...)` the error occurs. 
However, if we provide an array with the underlying files ourselves e.g. `read_json(['ndjson/file1.ndjson','ndjson/file2.ndjson', 'ndjson/file3.ndjson'],...)` everything works as expected.

**Comment by goober** [comment]
Yes, I can definitely test out the feature-branch when it is ready!

**Comment by goober** [comment]
@lnkuiper I have just tested your PR against our dataset and it is still providing the same error unfortunately.

**Comment by goober** [comment]
I ran it once again with a `make debug` build and got the following response:
>  Error: INTERNAL Error: Assertion triggered in file "temp/duckdb/src/common/types/string_type.cpp" on line 16: utf_type != UnicodeType::INVALID

**Comment by hannes** [comment]
Yes this looks pretty dramatic and will be fixed soon

**Comment by lnkuiper** [comment]
> Unfortunately I have not been able to reproduce the issue with a randomly generated dataset that can be shared. I will try to reproduce the issue with sharable data if necessary.

If this is somehow possible that would be of great help. If not I will try creating data sets using the schema you provided in the issue.

**Comment by goober** [comment]
@lnkuiper I will do another try to generate a dataset to reproduce the issue. Hopefully have some time during this week.

**Comment by lnkuiper** [comment]
@goober I recently fixed a string-related bug that affected the parquet writer (included in #7931). If you have time, could you check if the latest `master` fixes this issue?

**Comment by lnkuiper** [comment]
@goober Thanks for the quick reply, and thanks again for testing. Happy that it's fixed! 👍
